compiler/modules/front/stmt.bal

```diff
@@ -370,42 +361,51 @@ function addBindings(BindingChain? bindingLimit, BindingChain? bodyBindings, int
 }
 
 function codeGenStmt(StmtContext cx, bir:BasicBlock? curBlock, Environment env, s:Stmt stmt) returns CodeGenError|StmtEffect {
+    boolean scopedStmt = stmt is ScopedStmt;
```

Comment:
I think we can do this more cleanly by adding a function

```
function  codeGenScopedStmt(StmtContext cx, bir:BasicBlock curBlock, Environment env, ScopedStmt stmt) returns CodeGenError|StmtEffect {
    cx.pushScope(start.startPos, stmt.endPos);
    StmtEffect effect;
    if stmt is s:MatchStmt {
        effect = check codeGenMatchStmt(cx, curBlock, env, stmt);
    }
    // etc
    cx.popScope();
    return effect;
}
```

Reply:
fixed